rtgui/edit.h

```diff
@@ -152,6 +145,59 @@ class EditBuffer;
  *
  */
 
+
+
+class ObjectMOBuffer
+{
+private:
+
+    // Used to draw the objects where the color correspond to the object's ID, in order to find the correct object when hovering
+    Cairo::RefPtr<Cairo::ImageSurface> objectMap;
+    // If more than 254 objects has to be handled, objectMap2 contains the "upper part" of the 16 bit int value. objectMap2 will be NULL otherwise.
+    Cairo::RefPtr<Cairo::ImageSurface> objectMap2;
+    ObjectMode objectMode;
+
+protected:
+
+    // To avoid duplicated information, we points to a EditDataProvider that contains the current EditSubscriber
+    // instead of pointing to the EditSubscriber directly
+    EditDataProvider* dataProvider;
+
+    void createBuffer(int width, int height);
+    void resize(int newWidth, int newHeight, EditSubscriber* newSubscriber);
+    void flush();
+    EditSubscriber *getEditSubscriber ();
+
+public:
+    ObjectMOBuffer(EditDataProvider *dataProvider);
+    ~ObjectMOBuffer();
+
+    EditDataProvider* getDataProvider()
+    {
+        return dataProvider;
+    }
+    void setObjectMode(ObjectMode newType);
+    ObjectMode getObjectMode()
+    {
+        return objectMode;
+    }
+
+    Cairo::RefPtr<Cairo::ImageSurface> &getObjectMap ()
```

Comment:
If everybody has non-const reference access to the field, there is no use in declaring a getter since its type will also be unchangeable later on. Either access should be const or the fields can just become public as well.